### DISCUSSION TITLE
Is there a --read_datasink option for the Analysis step?

### DISCUSSION THREAD

**Discussion by geowk** [CoBrALab/RABIES#361] [General]
Ran into an issue, due to 1 bad datasets, during confound correction. Somehow it got through preprocessing. Looks like something weird happened with the warping to commonspace.

Confound correction continued to run throughout the remaining dataset, but at the end of the batch, gave an error and did not save a rabies_confound_correction_workflow.pkl due to that one bad dataset

Is there a way to run Analysis on this data or is there a --read_datasink option with Analysis to use? Trying to avoid having to re-run confound correction again for over 60 subjects.

-g

**Comment by Gab-D-G** [comment]
Hi, there are no --read_datasink option for analysis. You can try rerunning confound correction using the --exclusion_ids option to exclude the bad scan, and other subjects should not have to be run again through confoudn correction.

**Comment by geowk** [comment]
Thanks.

I tried using the option with the singularity container rabies 0.5.1.

When it says to add the full path to the txt or nii files, does this mean to add the path after the --exclusion_id  or with -B option. Tried just about every combination, and still receive the below message. 

```
singularity run -B ${sigmadir}/:${sigmadir}/ \
	    -B  ${rabies_dir}/${input}:/input_BIDS:ro \
	    -B  ${rabies_dir}/${preproc}:/preprocess_outputs/ \
	    -B ${rabies_dir}/${confcorr}:/confound_correction_outputs \
	    ${rabies_dir}/rabies.sif --exclusion_ids /data/rabies//input/sub-41/ses-02/func/sub-41_ses-02_task-rest_bold.nii.gz \
	    -p MultiProc --local_threads 5  --scale_min_memory 0.5 \
	    confound_correction /preprocess_outputs /confound_correction_outputs \
	    --image_scaling voxelwise_mean \
	    --highpass 0.009 \
	    --lowpass 0.2 \
	    --smoothing_filter 0.8 \
  	    --edge_cutoff 30 \
	     --read_datasink \
	    --ica_aroma apply=true,dim=0,random_seed=1
```


`ValueError: The --exclusion_ids ['/data/rabies//input/sub-41/ses-02/func/sub-41_ses-02_task-rest_bold.nii.gz']  input had improper format. It must the full path to a .txt or .nii files.`

**Reply by Gab-D-G** [reply]
The path probably needs to be accessible from inside the container, so can you try to provide a path that begins from /input_BIDS?

**Comment by geowk** [comment]
> The path probably needs to be accessible from inside the container, so can you try to provide a path that begins from /input_BIDS?

Okay. Thank you for your suggestion. That was the problem. Added the path with the -B option:

-B ${rabies_dir}/${input}/sub-41/ses-02/func/:${rabies_dir}/${input}/sub-41/ses-02/func/  \

```
singularity run -B ${sigmadir}/:${sigmadir}/ \
	    -B  ${rabies_dir}/${input}:/input_BIDS:ro \
            -B ${rabies_dir}/${input}/sub-41/ses-02/func/:${rabies_dir}/${input}/sub-41/ses-02/func/  \
	    -B  ${rabies_dir}/${preproc}:/preprocess_outputs/ \
	    -B ${rabies_dir}/${confcorr}:/confound_correction_outputs \
	    ${rabies_dir}/rabies.sif --exclusion_ids ${rabies_dir}/${input}/sub-41/ses-02/func/sub-41_ses-02_task-rest_bold.nii.gz \
	    -p MultiProc --local_threads 5  --scale_min_memory 0.5 \
	    confound_correction /preprocess_outputs /confound_correction_outputs \
	    --image_scaling voxelwise_mean \
	    --highpass 0.009 \
	    --lowpass 0.2 \
	    --smoothing_filter 0.8 \
  	    --edge_cutoff 30 \
	     --read_datasink \
	    --ica_aroma apply=true,dim=0,random_seed=1
```
After including this line it now works.

Thanks.

-g
